backend/danswer/llm/build.py

```diff
@@ -11,48 +9,40 @@
 from danswer.configs.model_configs import GEN_AI_TEMPERATURE
 from danswer.configs.model_configs import INTERNAL_MODEL_VERSION
 from danswer.direct_qa.qa_utils import get_gen_ai_api_key
-from danswer.dynamic_configs.interface import ConfigNotFoundError
 from danswer.llm.azure import AzureGPT
 from danswer.llm.google_colab_demo import GoogleColabDemo
 from danswer.llm.llm import LLM
 from danswer.llm.openai import OpenAIGPT
 
 
-def get_llm_from_model(model: str, **kwargs: Any) -> LLM:
-    if model == DanswerGenAIModel.OPENAI_CHAT.value:
-        if API_TYPE_OPENAI == "azure":
-            return AzureGPT(**kwargs)
-        return OpenAIGPT(**kwargs)
-    if (
-        model == DanswerGenAIModel.REQUEST.value
-        and kwargs.get("model_host_type") == ModelHostType.COLAB_DEMO
-    ):
-        return GoogleColabDemo(**kwargs)
-
-    raise ValueError(f"Unknown LLM model: {model}")
-
-
 def get_default_llm(
-    api_key: str | None = None, timeout: int = QA_TIMEOUT, **kwargs: Any
+    api_key: str | None = None,
+    timeout: int = QA_TIMEOUT,
 ) -> LLM:
     """NOTE: api_key/timeout must be a special args since we may want to check
     if an API key is valid for the default model setup OR we may want to use the
     default model with a different timeout specified."""
     if api_key is None:
-        try:
-            api_key = get_gen_ai_api_key()
-        except ConfigNotFoundError:
-            # if no API key is found, assume this model doesn't need one
-            pass
+        api_key = get_gen_ai_api_key()
+
+    model_args = {
+        # provide a dummy key since LangChain will throw an exception if not
+        # given, which would present server startup
+        "api_key": api_key or "dummy_api_key",
```

Comment:
Pretty sure there are cases where someone is using an openai compatible server and this should be empty.

Reply:
there should be no change in behavior from this diff - it would have be assigned the value of `dummy_api_key` in `OpenAIGPT` / `AzureGPT` anyways.

Also, I believe in most cases if the server doesn't use the key it will just be ignored

Reply:
ah true ok ok